compiler/src/dotty/tools/dotc/core/ConstraintHandling.scala

```diff
@@ -97,15 +97,21 @@ trait ConstraintHandling {
         constraint = constraint.replace(param, bound)
         true
       else
+        val dropWildcards = new ApproximatingTypeMap:
```

Comment:
We have a lot of logic for removing wildcards when instantiating a type variable, can it actually be removed with these changes? https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/dotc/core/ConstraintHandling.scala#L247-L309

Reply:
After some more tweaks it looks like the answer is yes.